crates/metal-llvm/src/lib.rs

```diff
@@ -1,22 +1,28 @@
 //! Metal library for compiling to LLVM IR using MIR.
 
-use metal_mir::parcel::Module;
+use std::collections::HashMap;
+
+use metal_mir::{parcel::Module, types::visibility::Visibility};
 
 pub mod expr;
 pub mod primitives;
 pub mod stmt;
 pub mod ty;
 
-use llvm_sys::prelude::{LLVMBuilderRef, LLVMContextRef, LLVMModuleRef, LLVMTypeRef, LLVMValueRef};
+use llvm_sys::{
+    prelude::{LLVMBuilderRef, LLVMContextRef, LLVMModuleRef, LLVMTypeRef, LLVMValueRef},
+    LLVMLinkage,
+};
 
 pub struct LLVMRefs {
     ctx: LLVMContextRef,
     builder: LLVMBuilderRef,
     module: LLVMModuleRef,
+    locals: HashMap<&'static str, LLVMValueRef>,
 }
 
 pub trait CodeGenValue {
-    fn codegen_value(&self, llvm: &LLVMRefs, module: &Module) -> LLVMValueRef;
+    fn codegen_value(&self, llvm: &mut LLVMRefs, module: &Module) -> LLVMValueRef;
```

Comment:
I can't quite find the reason for this change?

Reply:
`llvm.locals` (a hashmap of the local variables in a function) needs to be mutated in some CodeGenValues